### PR TITLE
Fixes for Mac build and crash on startup

### PR DESCRIPTION
**Pull request by hyperlogic**
For some reason the qt code gen is getting confused by the #if 0
in 3DConnextionClient.h, so I added a stub implementation.  This should fix this error:

    /Users/ajt/code/hifi/interface/src/devices/3DConnexionClient.h:0: Note: No relevant classes found. No output generated.
    Command /bin/sh emitted errors but did not return a nonzero exit code to indicate failure

In Application.cpp the change of moving idle into paintGL had a sideeffect
of calling OffscreenGlCanvas before it was initialized.  To work around this
I added a guard to prevent calling idle before all the gl windows/widgets have
been initialized.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 69f1cfbcb9125aa9d449a5d01b3734bcd51309e8 available at: https://gist.github.com/80602a343eed0c2171ee<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6308.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6308.dmg

**Comment by howard-stearns** [comment]
A Debug build from source (clean cmake and completely new .ini file) did not work for me: Crashed in updateRig->updateJointState->[] on startup. But maybe my build is screwed up. I'll try again later when I've got a steady internet connection.

**Comment by hifi-gustavo** [comment]
Build console output for commit: d46d521d81e65b62a47b13740cd94a8438a6e020 available at: https://gist.github.com/80602a343eed0c2171ee<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6308.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6308.dmg

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -1059,7 +1057,9 @@ void Application::paintGL() {
         _lastFramesPerSecondUpdate = now;
     }
 
-    idle(now);
+    if (_isGLInitialized) {
```
Comment (Atlante45) [inline interface/src/Application.cpp]: Shouldn't we bail from paintGL all-together if we haven't initialized yet?

File: interface/src/Application.h [hunk 2]
```diff
@@ -548,6 +548,7 @@ private slots:
     quint64 _lastSimsPerSecondUpdate = 0;
     bool _isForeground = true; // starts out assumed to be in foreground
     bool _inPaint = false;
+    bool _isGLInitialized {false};
```
Comment (Atlante45) [inline interface/src/Application.h]: Coding Standard: Spacing should be `Type var { val };`

**Comment by jherico** [comment]
This should be resolved now that we've migrated the spacemouse code into the input-plugins library.  the `#if 0` was a placeholder because the migration was in progress but changes in the controller infrastructure broke the existing spacemouse code.  This PR should no longer be required.

**Comment by howard-stearns** [comment]
Hmm. On the one hand, I've confirmed that this pr compiles (Debug scheme) but the result does not run. But on the other hand, master (with the Spacemouse migration #6286 merged) doesn't even compile at all on mac. I'm confused.

**Comment by hyperlogic** [comment]
@jherico The #6286 pull request does not build if HAVE_3DCONNEXIONCLIENT is not defined

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8d16f60b15ee192e07bfffa2bbc8b4c556e6b920 available at: https://gist.github.com/80602a343eed0c2171ee<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6308.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6308.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2318e48fa8dfce0e0ad2ecd1cf108693a57998f5 available at: https://gist.github.com/80602a343eed0c2171ee<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6308.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6308.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 50e8f49f5241820f364e9beeb783ab6731d160af available at: https://gist.github.com/80602a343eed0c2171ee<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6308.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6308.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 50370479e475fa1c3f54695a674e230f21c905e4 available at: https://gist.github.com/80602a343eed0c2171ee<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6308.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6308.dmg

**Comment by howard-stearns** [comment]
I've confirmed that:
 master does not build at all on mac
 this pr builds and runs for: mac/Debug + reset .ini; mac/Debug + reuse .ini; windows/release

There are other crashes, but I think these are independent of this release:
- assert failure on exit within deletion. Both platforms.
- intermittent entity script failure (e.g., in alpha space).

**Pull request merged**